execution/executor/src/lib.rs

```diff
@@ -521,9 +522,20 @@ where
 
         // Since other validators have committed these transactions, their status should all be
         // TransactionStatus::Keep.
-        for output in &vm_outputs {
-            if let TransactionStatus::Discard(_) = output.status() {
-                bail!("Syncing transactions that should be discarded.");
+        for (index, output) in vm_outputs.iter().enumerate() {
+            if let TransactionStatus::Discard(status_code) = output.status() {
+                let bail_error_message = format!(
+                    "Syncing a transaction that should be discarded! Transaction version: {:?}, status code: {:?}.",
+                    first_version + index as u64, status_code
+                );
+                error!("{}", bail_error_message);
+                info!("Discarded transaction: {:?}", transactions[index]);
+                info!("Discarded transaction output: {:?}", output);
+                info!(
+                    "Transactions were reexecuted in replay_transactions_impl: {:?}",
```

Comment:
nit: Could we update this to:

```
info!("Transactions were executed: {:?}", transactions_executed);
```

1. No need to provide the function name, our logging system supports code locations and backtraces 😄 
2. Technically, because we're state syncing, the transactions haven't been executed on this node before, so I think "re-executed" isn't that clear.